torch/csrc/Size.cpp

```diff
@@ -149,6 +149,26 @@ static PyObject* THPSize_repr(THPSize* self) {
   END_HANDLE_TH_ERRORS
 }
 
+static PyObject* THPSize_add(PyObject* self, PyObject* other) {
+  HANDLE_TH_ERRORS
+  if (!PyTuple_Check(other)) {
```

Comment:
You should also check that self is a subclass of tuple !

Reply:
Done. I also added that test that checks that concatenation with non-tuple types fails with `TypeError`. Without this check, it actually seg-faulted.